examples/performance/00_overview.ipynb

```diff
@@ -0,0 +1,753 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "op2_omp = Operator(eq, opt=('blocking', 'simd', {'openmp': True}))\n",
+    "# print(op2_omp)  # Uncomment to see the generated code\n",
+    "# print(_unidiff_output(str(op1_omp), str(op2_omp)))  # Uncomment to print out the diff only"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Code motion\n",
+    "\n",
+    "The `advanced` mode performs a code motion pass. In explicit PDE solvers, this is most commonly used to lift expensive time-invariant sub-expressions out of the inner loops. The pass is however quite general in that it is not restricted to the concept of time-invariance -- any sub-expression invariant with respect to a subset of `Dimension`s is a code motion candidate. In the example below, we see that `sin(f)` gets hoisted out of the inner loops since it is determined to be an expensive invariant sub-expression. In other words, the compiler trades the redundant computation of `sin(f)` for additional storage (the `r1[...]` array)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "op3_omp = Operator(eq, opt=('lift', {'openmp': True}))\n",
+    "_print_kernel(op3_omp)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Basic flop-reducing transformations\n",
+    "\n",
+    "Among the simpler flop-reducing transformations applied by the `advanved` mode we find:\n",
+    "\n",
+    "* \"classic\" common sub-expressions elimination (CSE),\n",
+    "* factorization,\n",
+    "* optimization of powers\n",
+    "\n",
+    "The cell below shows how the computation of `u` changes by incrementally applying these three passes. First of all, we observe how the symbolic spacing `h_y` gets assigned to a temporary, `r0`, due to appearing in several sub-expressions. This is the effect of CSE."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "op4_omp = Operator(eq, opt=('cse', {'openmp': True}))\n",
+    "print(_unidiff_output(str(op0_omp), str(op4_omp)))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Factorization causes the collection of `r0` in two different sub-expressions. In larger, more complex examples, the impact of factorization is typically much more pronounced."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "op5_omp = Operator(eq, opt=('cse', 'factorize', {'openmp': True}))\n",
+    "print(_unidiff_output(str(op5_omp), str(op4_omp)))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "There are no powers in this example, so the `opt-pows` pass has no effect."
```

Comment:
Why not adding some to the eq? TO show that as well?

Reply:
good point, doing it